src/aerovaldb/utils/string_utils.py

```diff
@@ -4,21 +4,22 @@
 PATH_COMPONENT_PATTERN = re.compile(r"^[^/]+$", flags=re.UNICODE)
 
 
-def str_to_bool(value: str, /, strict: bool = False, default: bool = False) -> bool:
+def str_to_bool(value: str, /, default: bool | None = None) -> bool:
     """
     Parses a string as a boolean, supporting various values. It is intended
     mainly for parsing environment variables.
 
     Supports 1/0, true/false, t/f, yes/no, y/n (Case insensitive).
 
     :param value : The string value to be converted.
-    :param strict : If true, ValueError is raised if an unrecognized value is encountered,
-    otherwise default is returned.
-    :param default : Returned if strict is disabled, for unrecognized values.
+    :param default : If None, a string value not explicitly matching one of the
+        above values for true/false will raise a ValueError. Otherwise this default
+        will be returned instead.
+
```

Comment:
You document here foremost an internal value (None) which the user is not supposed to set. I suggest: 
```
Optional default return value (True/False) if the string value doesn't match any supported value. If not set, or set to None, a ValueError is raised in these cases.
```